### DISCUSSION TITLE
通过定制化导出功能支持导入到任意双链笔记（如：将标注导入到 Obsidian 中）

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2222] [服务]
> **[简悦 2.2.0](http://ksria.com/simpread/welcome/version_2.2.0.html)** 发布后，增加了对 Markdown 定制化功能，通过此功能可以方便将标注 → 定制化 Markdown → 其它双链笔记

## 前提

> Markdown 模板语法请看 [定制化导出](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=markdown)，同时使用此功能的用户一般都需要 [配置知识库](https://github.com/Kenshin/simpread/discussions/2221)。

## 其它说明

> 注意：此功能 **仅支持标注的定制化** 时生效，也就是说，如果你想 **定制化全文的话，暂时没有提供此功能**。

## 模板库

> 简悦官方维护一个 [模板库](https://github.com/Kenshin/simpread/discussions/2153)，里面有来自官方与社区用户的提供的大量模板，可供使用。

## 配置全文

> 在 https://github.com/Kenshin/simpread/discussions/2153 随便找一个模板，如 https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-756801 将此内容复制到下面截图中，位置在：选项页 → 服务 → 定制导出

![image](https://user-images.githubusercontent.com/81074/119213221-20caa180-baf0-11eb-9a50-af9c8ba8e0fd.png)

> 同时下面需要注意需要添加的内容，因为是教程，我直接写出结论，如果希望知道这些是什么的话，[请看这里](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=%e5%85%b6%e5%ae%83%e7%9a%84%e4%b8%80%e4%ba%9b%e5%ae%9a%e5%88%b6)

![image](https://user-images.githubusercontent.com/81074/119213379-31c7e280-baf1-11eb-9645-23a7ed6877f0.png)

## 配置单条标注

> 在 https://github.com/Kenshin/simpread/discussions/2153 随便找一个模板，如 https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-757014 将此内容复制到下面截图中，位置在：选项页 → 服务 → 定制导出

![image](https://user-images.githubusercontent.com/81074/119213431-8f5c2f00-baf1-11eb-87d5-82d9c5ce270b.png)

## 配置浮动工具条的导出

> 位置在选项页 → 高级设定 → 标注

![image](https://user-images.githubusercontent.com/81074/123213551-b47cfc80-d4f8-11eb-97c6-da0036852943.png)

## 配置完成

> 通过以上的方式，就完成了配置。

> 下面举例说明一些场景。

## 例子一

> 将单条标注复制到剪切板 → 粘贴到任意支持双链笔记里面，同时可以打开 **永久链接**，关于 **永久链接** 的说明，可以看 [当简悦变成你的知识库（离线且具有永久链接） ](https://github.com/Kenshin/simpread/discussions/2221)，此功能需要配合同步助手，关于这方面的教程可以 [看这里](https://github.com/Kenshin/simpread/discussions/2147)。

> 关于此功能的更多细节介绍可以看 [利用好浮动标注栏的导出功能](https://github.com/Kenshin/simpread/discussions/1851)。

![2021-05-22_11-54-22 (1)](https://user-images.githubusercontent.com/81074/119213824-a9e3d780-baf4-11eb-8be2-eb4fc68a03c6.gif)

## 例子二

> 在**阅读模式下导出全文**

![image](https://user-images.githubusercontent.com/81074/123214443-cd39e200-d4f9-11eb-8a95-482adfad0ddd.png)

> 粘贴后的效果（Obsidian）

![image](https://user-images.githubusercontent.com/81074/123214498-e2167580-d4f9-11eb-9720-228337a0fbd4.png)

## 例子三

> 改为导出全部标注，这次是在**稍后读的阅读模式**中操作，功能点：

- 标注的本地实时更新（ 标注也保存在本地，同时可以在浏览器里访问
- 复制定制化的 Markdown 到剪切板
- 粘贴到 Obsidian
- 在浏览器打开永久链接

![2021-05-22_12-25-50 (2)](https://user-images.githubusercontent.com/81074/119214454-8cfdd300-baf9-11eb-975d-2ae9298c3714.gif)

## 例子四

> 在**稍后读标注模式**下使用

![image](https://user-images.githubusercontent.com/81074/123214829-46d1d000-d4fa-11eb-8b03-05419a8d8fd1.png)

#### **仅导出标注**

![image](https://user-images.githubusercontent.com/81074/123215063-8e585c00-d4fa-11eb-8a83-249065524155.png)

![image](https://user-images.githubusercontent.com/81074/123215158-af20b180-d4fa-11eb-947a-d9f6d6965025.png)

![image](https://user-images.githubusercontent.com/81074/123215119-a203c280-d4fa-11eb-8f5b-24dccac6df6f.png)

#### **导出全文**

![image](https://user-images.githubusercontent.com/81074/123215002-77196e80-d4fa-11eb-8929-e58871bf87c2.png)

> 上述只是基础用法，也就是说，导出的 Markdown 结构是固定的，不能因人而异。

## 引申

只要通过 [定制化导出](http://ksria.com/simpread/docs/#/定制化导出?id=markdown) 的定制不同的双链笔记可以识别的结构，都可以实现，不仅只是 Obsidian。

简悦稍后读内的任意内容均可导出到生产力工具，包括：[双向链接](http://ksria.com/simpread/docs/#/%E7%A8%8D%E5%90%8E%E8%AF%BB-%E5%A4%9A%E7%A7%8D%E5%B8%83%E5%B1%80?id=%e5%8f%8c%e5%90%91%e9%93%be%e6%8e%a5)，所以你可以通过此方式设置更多导出效果。

简悦支持 [永久链接](https://github.com/Kenshin/simpread/discussions/3188) 你可以在任意双链笔记中使用此方式作为标注的原出处。

## 一个更方便的例子

> Obsidian 与简悦一样，都是基于本地系统，所以上面的 **例子三** 可以这么做：

1. 配置 简悦 · 同步助手的 **[增强导出功能](http://ksria.com/simpread/docs/#/Sync?id=%e5%af%bc%e5%87%ba%e6%9c%8d%e5%8a%a1)**

2. Obsidian 的 Vault 设置为 同步助手的导出文件夹，假设叫 `output`

3. 配置 **[当产生标注时，如何自动同步到本地](https://github.com/Kenshin/simpread/discussions/2220)** 

4. 在 **阅读模式 或 稍后读** 里面，产生标注后，就自动存在与 `output` 文件夹，并且可以被 Obsidian 直接识别并使用。

![image](https://user-images.githubusercontent.com/81074/119214696-43ae8300-bafb-11eb-8a3d-876a624eb68a.png)

## 更多例子

> 关于定制化导出更多例子可以 [看这里](https://github.com/Kenshin/simpread/discussions?discussions_q=label%3Amdtemplate)

- [定制导入到 flomo 的模板](https://github.com/Kenshin/simpread/discussions/2275)

- [利用好浮动标注栏的导出功能](https://github.com/Kenshin/simpread/discussions/1851)

- [借助定制化 Markdown 导出标注的纯文本信息](https://github.com/Kenshin/simpread/discussions/2277)

## 参考知识点

> 上述操作都已经写到了文档里面，有感兴趣的用户，可以看下。

- 简悦 · 同步助手的 [增强导出功能](http://ksria.com/simpread/docs/#/Sync?id=%e5%af%bc%e5%87%ba%e6%9c%8d%e5%8a%a1)
- [自定义标题](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=%e8%87%aa%e5%ae%9a%e4%b9%89%e5%af%bc%e5%87%ba%e6%a0%87%e9%a2%98)
- [自动化](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%8C%96)
- [模板参数说明](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=markdown)

**Comment by hellozgl** [comment]
请教一下，如何将多个标注一次性复制到obsidian中（我是用最原始的方法，即按钮“复制markwodn到剪切板”，再clrl v到obsidian）。
目前已经设置了自定义markdown模版，单条导出和多条导出都已经自定义了模板格式，模板来源是从您提供的几个模板中复制下来的。
测试下来，通过浮动标注栏的单条导出功能（下图3），单条导出到obsidian是没有问题的，但当我想把页面上多个高亮的标注导出到obsidian中，发现没有多个高亮导出的按钮，而在稍后读的页面，我若是选择“复制markdown到剪切板”按钮（下图5），复制到obsidian中，却是将整个网页的所有内容，而不仅仅是标注，变成markdown格式，请问该如何导出多条标注，以markdown导出到obsidian中呢？
![截屏2021-11-17 21 41 35](https://user-images.githubusercontent.com/70676660/142213269-67419e68-6f99-4752-9569-19533b957a60.png)
![截屏2021-11-17 21 41 53](https://user-images.githubusercontent.com/70676660/142213288-53b2bf7b-4cb8-435e-b62f-9343f4eb2b4b.png)
![截屏2021-11-17 21 49 04](https://user-images.githubusercontent.com/70676660/142213291-4d967505-4b62-423e-9c89-f345589b4ad7.png)
![截屏2021-11-17 21 51 48](https://user-images.githubusercontent.com/70676660/142213297-0eb53e24-1be6-4605-87d7-69d9a5a12963.png)
![截屏2021-11-17 22 13 07](https://user-images.githubusercontent.com/70676660/142216901-6561bd53-b96b-48c2-87e2-78abf1970fa4.png)

**Comment by hellozgl** [comment]
我刚刚尝试了设置：选项页 → 高级设定 →导出时可选→选“仅标注”（原来选的是“全文（含标注）+标注”），这样操作以后，可以实现只复制出高亮内容为markdown格式，黏贴在obsidian中只显示了高亮内容。如下图所示。
但产生了新的问题：由于没有自动离线备份全文，仅仅自动离线备份了高亮内容的.md文件，这样我在obsidian中就无法离线读取高亮部分对应的原文了（在html格式下，高亮的内容是可以自动关联到离线原文的位置的），有什么办法可以使markdown格式下实现与html相同的一一原文对应功能吗
![截屏2021-11-17 23 11 18](https://user-images.githubusercontent.com/70676660/142227412-0a682d85-3d94-46dc-bbc5-a6b1703afb27.png)

**Reply by Kenshin** [reply]
嗨，你的问题可以很简单的解决：使用 [自动化](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%8C%96)，同时导出 HTML 和 Markdown 就可以了，但请选择 `全文（含标注）+标注`，这样当产生标注时，会生成四个文件。

1. 两个 html e.g. `123-xxxx.html`  和`123-xxx@annote.html`
2. 两个 md 文件，同上

**Comment by willisfusu** [comment]
你好 Kenshin，

我发现一个问题：

如果网页标题中有「空格」，那么使用Markdown模板，占位符{{an_int_uri}} 生成的链接中，「空格」不会转换为“％20”。

这样产生一个问题，即Markdown文件中不能正确识别该链接。

我想，这个问题，如果能对{{an_int_uri}}加一个超链接格式化的操作，应该会避免。或者是有其它解决方法嘛？

谢谢。

**Reply by willisfusu** [reply]
补充图片
![image](https://user-images.githubusercontent.com/32933353/147265269-eba0aff0-9d9b-4912-bc1c-fbf3608e5869.png)

![image](https://user-images.githubusercontent.com/32933353/147265377-7be28846-e467-43a2-9d56-32f7007f80cd.png)

**Reply by Kenshin** [reply]
试试这样的方式

`[内部链接](<{{an_int_uri}}>)`

**Reply by willisfusu** [reply]
It does work!  Thanks.

**Reply by mtuwei** [reply]
> 试试这样的方式
> 
> `[内部链接](<{{an_int_uri}}>)`

Kenshin，你好！发现一个新问题：如果在【自定义标题】中设置了【{{id}}{{un_title}}】，使用{{an_int_uri}}导出的链接不会包含{{id}}。因此，不能正常跳转，原因似乎也在于html文件的标题多了【{{id}}-】。
由于{{id}}是【稍后读】的占位符，我尝试放在【标记】导出模板中，但目前似乎也不支持。因此，请问如何能解决这个问题，谢谢！

**Comment by HarveyZed** [comment]
你好，遇到一个问题
定制化导出date_format失败
按说明文档加空格，但在}}后面加或者在now和id后面加都没起作用，请帮我看看
```
---
title: Obsidian 的 YAML Front matter 介绍@annote
date: {{date_format|now |yyyy-MM-dd hh:mm:ss}}   
time: {{date_format|id  |yyyy-MM-dd hh:mm:ss}}  
create: 2022年01月18日 19:00:26
tags: 
- 简悦
- 标注
from: Obsidian 的 YAML Front matter 介绍
host: zhuanlan.zhihu.com
url: https://zhuanlan.zhihu.com/p/370113792
category: notes
alias: 
---
date: {{date_format|now |yyyy-MM-dd hh:mm:ss}}  
time: {{date_format|id  |yyyy-MM-dd hh:mm:ss}}  
```

**Comment by tanghui123th** [comment]
你好，我遇到一个问题，backlinks感觉是失效的不知道是哪配置的问题
![](https://pic-mgmt.oss-cn-shanghai.aliyuncs.com/img/20220218000521.png)
![](https://pic-mgmt.oss-cn-shanghai.aliyuncs.com/img/20220218000641.png)

**Reply by Kenshin** [reply]
`{{backlinks}}` 需要改成 `{{  |an_backlinks}}`

**Reply by Kenshin** [reply]
补充下，这个算是个 Bug，虽然不影响什么，下个版本会让此方式更灵活。

**Reply by tanghui123th** [reply]
好的，等下一个版本修复。目前{{backlinks}}和 {{ |an_backlinks}}都没有作用

**Reply by Kenshin** [reply]
应该没问题的，你把你的模板发过来，我看下。

也可以根据 https://github.com/Kenshin/simpread/discussions/2153 里面已定制好的模板，自己试试。（通过检索关键字 `backlinks` 看看都是怎么写的和设计的。

**Reply by tanghui123th** [reply]
> 应该没问题的，你把你的模板发过来，我看下。
> 
> 也可以根据 #2153 里面已定制好的模板，自己试试。（通过检索关键字 `backlinks` 看看都是怎么写的和设计的。

首先，不知道我的设置对不对
![](https://pic-mgmt.oss-cn-shanghai.aliyuncs.com/img/20220219101418.png)


## backlinks试过三种方式
```
第一种：- {{ |backlinks}}
第二种：- {{backlinks}}
第三种：- {{ |an_backlinks}}
```
## 第一种方式，模板如下：

```
title:: 文摘-{{title}}
tags:: SimpRead

## Meta Data
        - Title:: {{title}}
          Source:: {{url}}
          Date:: [[{{date_format|now|yyyy-MM-dd}} ]]
          Summary:: "{{desc}}"
## 关联文章
        - {{ |backlinks}}
## 摘要
{{#each}}
        - [📌](<{{an_int_uri}}>)  {{an_html}}
{{>|an_note}}{{an_tags}}
{{/each}}

```
得到的结果：
![](https://pic-mgmt.oss-cn-shanghai.aliyuncs.com/img/20220219100801.png)

## 第二种方式得到的结果
![](https://pic-mgmt.oss-cn-shanghai.aliyuncs.com/img/20220219102633.png)
logseq中的效果
![](https://pic-mgmt.oss-cn-shanghai.aliyuncs.com/img/20220219102937.png)

## 第三种方式得到的结果
![](https://pic-mgmt.oss-cn-shanghai.aliyuncs.com/img/20220219102323.png)

**Reply by Kenshin** [reply]
emmm 我好像知道你的问题了。

简悦的反向链接是需要在简悦中操作的，看下图

![image](https://user-images.githubusercontent.com/81074/154792147-41f77ed5-4747-45cc-b72f-5ee51540b6b1.png)

第一种和第三种是正确的，但第三种需要放在 `each` 的中间才能生效

![image](https://user-images.githubusercontent.com/81074/154792218-457acf75-f740-4bd9-8958-6c3c2b46f94c.png)
